File: app/src/main/java/com/erikzuo/portfolioandroidapp/ui/main/education/EducationFragmentModule.java
package com.erikzuo.portfolioandroidapp.ui.main.education;

import android.support.v7.widget.LinearLayoutManager;

import com.erikzuo.portfolioandroidapp.data.model.Education;
import com.erikzuo.portfolioandroidapp.data.model.Repo;
import com.erikzuo.portfolioandroidapp.ui.main.work.personal.PersonalWorkFragment;

import java.util.ArrayList;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Soprano on 9/02/2018.
 */

@Module
public class EducationFragmentModule {

    @Provides
    EducationAdapter provideEducationAdapter() {
        return new EducationAdapter(new ArrayList<>());
    }

    @Provides
    LinearLayoutManager provideLinearLayoutManager(EducationFragment fragment) {
        return new LinearLayoutManager(fragment.getActivity());
    }
}

File: app/src/main/java/com/erikzuo/portfolioandroidapp/ui/main/education/EducationFragmentModule.kt
package com.erikzuo.portfolioandroidapp.ui.main.education

import android.support.v7.widget.LinearLayoutManager
import com.erikzuo.portfolioandroidapp.data.model.Education

import java.util.ArrayList

import dagger.Module
import dagger.Provides

/**
 * Created by YifanZuo on 9/02/2018.
 */

@Module
class EducationFragmentModule {

    @Provides
    internal fun provideEducationAdapter(): EducationAdapter {
        return EducationAdapter(ArrayList<Education>())
    }

    @Provides
    internal fun provideLinearLayoutManager(fragment: EducationFragment): LinearLayoutManager {
        return LinearLayoutManager(fragment.activity)
    }
}
